src/ProgressOnderwijsUtils/Html/HtmlFragment.cs

```diff
@@ -51,7 +51,7 @@ public void Deconstruct(out string textContent, out IHtmlElement element, out Ht
         }
 
         public bool IsEmpty
-            => Implementation == null;
+            => Implementation == null || Implementation is string implementation && string.IsNullOrEmpty(implementation);

```

Comment:
Beter om te zorgen dat dit niet eens voor kan komen, dus bij het vullen van `Implementation` dit checken.

Tevens is de code die hiervan afhankelijk is best tricky, wie weet is het ook handig daar iets mee te doen.